_posts/2019-02-01-dynamic-spawn-scheduler.md

```diff
@@ -71,14 +83,14 @@ iOS | iMac Pro 2018, 18-core | 34s | 414s ↑ | 32s ≈
 iOS | Mac Pro 2013, 6-core | 39s | 414s ↑ | 36s ≈
 iOS | MacBook Pro 2015, 4-core | 39s | 450s ↑ | 36s ≈
 
-As you can see, the remote-only configuration gave us slightly better results than the local-only configuration for clean builds, but the remote-only configuration became unacceptably slow for incremental builds. As expected, though, dynamic execution wins heads-down in all cases: clean builds are many times faster than they are with local or remote execution alone, and incremental builds are essentially the same.
-
-We expect the speedup to be more prominent as builds get larger, as seen in the difference between the iOS and Android builds.
+As you can see, the remote-only configuration gave us better results than the local-only configuration for clean builds, but the remote-only configuration became unacceptably slow for incremental builds. As expected, though, dynamic execution won heads-down in all cases: clean builds were many times faster than they are with local or remote execution alone, and incremental builds were essentially the same.
 
 ## Future work and credits
 
-There still are some problems to be resolved, of course. The major one is that local workers do not support cancellations ([#614](https://github.com/bazelbuild/bazel/issues/614])), and this can result in a high build timing variance depending on how the actions are scheduled. Other minor issues include: an artificial delay before running actions locally ([#7327](https://github.com/bazelbuild/bazel/issues/7327)) and making sure remote caches are populated even when we cancel actions ([#7328](https://github.com/bazelbuild/bazel/issues/7328)).
+There still are some problems to be resolved, of course, and is why this feature is still **experimental**. The major one is that *using this feature can lead to correctness issues unless you can guarantee that the local build environment matches the remote one* (if you cannot do this, avoid using this feature). Minor issues include the fact that local workers do not support cancellations ([#614](https://github.com/bazelbuild/bazel/issues/614])); high build timing variance depending on how the actions are scheduled; an artificial delay before running actions locally ([#7327](https://github.com/bazelbuild/bazel/issues/7327)); and making sure remote caches are populated even when we cancel actions ([#7328](https://github.com/bazelbuild/bazel/issues/7328)).
```

Comment:
It's unclear from the header that this section is really about caveats. What do you think about labeling it "Caveats" or "Considerations before using"? It also seems like credits could be a different section.

Reply:
Renamed and added credits. I don't really like such a section at the end but can't think of a better title.